notebooks/official/workbench/spark/spark_sample_notebook.ipynb

```diff
@@ -0,0 +1,1200 @@
+    "\n",
+    "poly_ranking.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "633a0752b053"
+   },
+   "source": [
+    "The majority of results contain a combination of HTML or CSS, and Javascript.\n",
+    "\n",
+    "Display a pie chart?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "30ae2d750ab7"
+   },
+   "outputs": [],
+   "source": [
+    "# Convert the Spark DataFrame to a Pandas DataFrame to display the pie chart.\n",
+    "poly_panda = poly_ranking.toPandas()[:LIMIT].copy()\n",
+    "poly_panda.groupby([\"poly_language\"]).sum().plot(\n",
+    "    kind=\"pie\",\n",
+    "    y=\"count\",\n",
+    "    autopct=\"%1.1f%%\",\n",
+    "    label=\"\",\n",
+    "    title=\"Polyglot repositories\",\n",
+    "    legend=False,\n",
+    "    figsize=(7, 7),\n",
+    "    explode=EXPLODE_PIE_CHART,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "3905037bcea3"
+   },
+   "source": [
+    "The pie chart shows that eight out of the top ten results contain either `HTML` or `CSS`.\n",
+    "\n",
+    "You can create combinations of languages in each repo using the original data fetched from BigQuery."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "25479dcebd93"
+   },
+   "outputs": [],
+   "source": [
+    "# A Python package to get combinations.\n",
+    "from itertools import combinations\n",
+    "\n",
+    "# A Python package to use type hint\n",
+    "from typing import List\n",
+    "\n",
+    "# PySpark functions\n",
+    "from pyspark.sql.functions import explode"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "ad6a3e548fc3"
+   },
+   "outputs": [],
+   "source": [
+    "def normalize_name(name: str) -> str:\n",
+    "    \"\"\"\n",
+    "    Change the language name to avoid invalid characters in the BigQuery data.\n",
+    "    Args:\n",
+    "        name: string\n",
+    "    Returns:\n",
+    "        Normalized name: string\n",
+    "    \"\"\"\n",
+    "    normalized_arr = []\n",
+    "\n",
+    "    # The following set of characters cannot be used in BigQuery's field.\n",
+    "    invalid_chars = {\",\", \";\", \"{\", \"}\", \"(\", \")\", \"\\n\", \"\\t\", \"=\"}\n",
+    "\n",
+    "    # The name must start with a letter or underscore.\n",
+    "    if name[0].isnumeric():\n",
+    "        normalized_arr.append(\"_\")\n",
+    "\n",
+    "    for ch in name:\n",
+    "        # Skip if a character is in the set of invalid characters.\n",
+    "        if ch in invalid_chars:\n",
+    "            continue\n",
+    "\n",
+    "        # Convert a space or dot to underscore\n",
+    "        elif ch == \" \" or ch == \".\":\n",
+    "            normalized_arr.append(\"_\")\n",
+    "\n",
+    "        # Change to lowercase to merge name duplications (for example, \"Java\" and \"java\")\n",
```

Comment:
duplications (for example, \"Java\" and \"java\") -> duplications, for example, \"Java\" and \"java\")